kubernetes/client/api_client.py

```diff
@@ -492,6 +492,21 @@ def select_header_content_type(self, content_types):
 
         content_types = [x.lower() for x in content_types]
 
+        if method and body and method == 'PATCH':
+            # try to select valid content_type
+            if 'application/json-patch+json' in content_types and \
+               'application/merge-patch+json' in content_types:
+                if isinstance(body, list):
+                    return 'application/json-patch+json'
+                else:
+                    # ---
+                    # kubernetes specific - application/strategic-merge-patch+json
+                    # applied as a patch or subclass
+                    if 'application/strategic-merge-patch+json' in content_types:
+                        return 'application/strategic-merge-patch+json'
+                    # ---
+                    return 'application/merge-patch+json'
```

Comment:
Most native types support both strategic merge patch and JSON merge patch. Just to clarify, if a client wants to send JSON merge patch, it has to use the `content_type` parameter to hint, right?

Reply:
Yes, it's hard to detect which type of patch has been sent so it's better to expose content_type explicit.